backend/app/routers/datasets.py

```diff
@@ -516,3 +519,44 @@ async def download_dataset(
         )
     else:
         raise HTTPException(status_code=404, detail=f"Dataset {dataset_id} not found")
+
+@router.post("/{dataset_id}/extract")
+async def get_dataset_extract(
+    dataset_id: str,
+    info: Request,
+    db: MongoClient = Depends(dependencies.get_db),
+    rabbitmq_client: BlockingChannel = Depends(dependencies.get_rabbitmq),
+):
+    if (dataset := await db["datasets"].find_one({"_id": ObjectId(dataset_id)})) is not None:
+        req_info = await info.json()
+        if 'extractor' in req_info:
+            # TODO check of extractor exists
+            msg = {"message": "testing", "dataseet_id": dataset_id}
+            body = {}
+            body['host'] = 'http://127.0.0.1:8000'
+            body['secretKey'] = 'secretKey'
+            body['retry_count'] = 0
+            body['filename'] = dataset['name']
+            body['id'] = dataset_id
+            body['datasetId'] = dataset_id
+            body['host'] = 'http://127.0.0.1:8000'
+            body['secretKey'] = 'secretKey'
+            body['resource_type'] = 'dataset'
+            body['flags'] = ""
```

Comment:
I'm not sure about if we should spell out the secretKey and host here...

Reply:
I took that out and used token, to match what happens with files.
So far I haven't actually tested dataset extraction, so this may change again later.